### DISCUSSION TITLE
Documentation links of referenced projects (nugets)

### DISCUSSION THREAD

**Discussion by vougkum** [Doraku/DefaultDocumentation#103] [Q&A]
Hi, the project which I am working on the documentation has a solution with library projects with package reference. I have configured each of them as below:
```
<GenerateDocumentationFile>true</GenerateDocumentationFile>
    <DefaultDocumentationFolder>..\..\..\docs\wiki\xx.yy.Core</DefaultDocumentationFolder>
    <DefaultDocumentationFileNameFactory>Name</DefaultDocumentationFileNameFactory>
    <DefaultDocumentationAssemblyPageName>xx.yy.Core</DefaultDocumentationAssemblyPageName>
    <DefaultDocumentationGeneratedPages>Assembly, Interfaces, Classes</DefaultDocumentationGeneratedPages>
    <DefaultDocumentationGeneratedAccessModifiers>Public</DefaultDocumentationGeneratedAccessModifiers>
    <DefaultDocumentationIncludeUndocumentedItems>true</DefaultDocumentationIncludeUndocumentedItems>
     <DefaultDocumentationRemoveFileExtensionFromLinks>true</DefaultDocumentationRemoveFileExtensionFromLinks>
    <DefaultDocumentationLinksOutputFile>..\..\..\docs\wiki\xx.yy.Core.Links.txt</DefaultDocumentationLinksOutputFile>
     <DefaultDocumentationExternLinksFiles>..\..\..\docs\wiki\*.Links.txt</DefaultDocumentationExternLinksFiles>
     <DefaultDocumentationLinksBaseUrl>https://dev.azure.com/..Wiki?pagePath=/xx.yy.Core/</DefaultDocumentationLinksBaseUrl>
```
But in the generated documentation the links to the referenced library are the default ones (https://docs.microsoft.com/en-us/dotnet/api/..) and not point to the link in azure (where my wiki is). When I change the solution to project reference instead of package reference the links are generated correctly. 
So I am guessing that package reference is not supported or am I missing something in the configuration?
I tried to use the <see cref="T:xx.yy.zz"> which generates a second link (with the correct url) but it also generates a link which points to docs.microsoft.com. 
For example in code: <param name="zz"><see cref="T:xx.yy.zz"/></param> I get two links zz(docs.microsoft.com) zz (dev.azure.com)
Can you please advice on this?

The tool is great, the documentation generation on build is a great feature and the fact that it generates also markdown files 
makes it perfect to publish it as a dev azure wiki. 

Many thanks

**Comment by Doraku** [comment]
hello, hum that's weird can you confirm that the xx.yy.Core.Links.txt is correctly produced? You can look for `ExternLinksFiles:` in the build logs to see if it correctly found all your links file.
Can you add `<DefaultDocumentationLogLevel>Debug</DefaultDocumentationLogLevel>` to your csproj properties and look into the build logs for one of the id that you expect from the link files?

**Reply by vougkum** [reply]
Hello again,
yes I can see in debug that it finds the right link files:
1>2022-03-15 15:26:06.6963|INFO|DefaultDocumentation|ExternLinksFiles:
1>	C:\QCS Projects\Platform\docs\wiki\Platform.Core.Links.txt
1>	C:\QCS Projects\Platform\docs\wiki\Platform.DataTransfer.Links.txt

I see in Platform.Core.Links.txt file the member that the other project references:
T:Platform.Globalization.ITimezoneService|Platform_Globalization_ITimezoneService|ITimezoneService
I also see the dev.azure.com url on the top of the link file. So I believe that Links.txt is correctly produced.

Platform.Globalization is a namespace in Platform.Core project and when I build a project that Platform.Core is a nested reference, for example Platform.Data has a dependency on Platform.DataTransfer which has a dependency on Platform.Core.
and contains a class with constructor:
```
public DataAccesServiceFactory(IEnvironmentConfigurationProvider environmentConfigurationProvider, IIdentityService identityService, ISqlParser sqlParser, **ITimezoneService timezoneService,** Func<string, int, ISqlParser, ITimezoneService, T> constructor)
      {
        ..
      }
```
the outcome is
![documentation_page](https://user-images.githubusercontent.com/61543130/158395263-8bcfdbfb-7bd1-4063-a9d4-216ceca42834.PNG)

Maybe the problem is with nested dependencies?

**Reply by Doraku** [reply]
The assembly loader should load indirect references without any issue, it would not be able to display the parameters type like so if it wasn't the case.
Do you see a `adding DocItem "DefaultDocumentation.Model.ExternDocItem" with id "T:Platform.Globalization.ITimezoneService"` log when you set the log level to `Debug`?

**Reply by vougkum** [reply]
No it is adding DocItem "DefaultDocumentation.Models.Types.InterfaceDocItem instead:
1>2022-03-15 16:31:15.9547|DEBUG|DefaultDocumentation|adding DocItem "DefaultDocumentation.Models.Types.InterfaceDocItem" with id "T:Platform.Globalization.ITimezoneService"
1>2022-03-15 16:31:16.3298|DEBUG|DefaultDocumentation|Writing DocItem "DefaultDocumentation.Models.Types.InterfaceDocItem" with id "T:Platform.Globalization.ITimezoneService"

**Reply by Doraku** [reply]
wait is that happening in the Platform.Core project or Platform.Data/Platform.DataTransfert?

**Reply by vougkum** [reply]
this is on Platform.Core build. sorry
when I build Platform.Data I see 
1>2022-03-15 17:10:16.7306|DEBUG|DefaultDocumentation|adding DocItem "DefaultDocumentation.Models.ExternDocItem" with id "T:Platform.Globalization.ITimezoneService"

**Reply by Doraku** [reply]
Rah yes ok, I remember! I actually fixed this bug last week >_> Any link you would have in your documentation with `<see cref="...">` to external type would work but the link generated from type introspection (like here the type of a parameter) would generate a bogus id and it could not find the ExternDocItem, thus falling back to the default dotnetapi link.
(https://github.com/Doraku/DefaultDocumentation/commit/eb8dbcb2a646baf917ac793d41c2ef36f2b051a7#diff-64f8e7a0ef27d88ca3795c2aec106fdb5cc90f6859ba318336b1f4e3e227de06R164 if you are curious)
The next version (with this fix and numerous new features) should be coming soon, I have written half of the new readme yesterday and still need to finish it. So hopefully by the end of the week it should be there :)

**Reply by vougkum** [reply]
many thanks!!looking forward to it!

**Comment by Doraku** [comment]
cleaning up old issues/discussion to keep relevant stuff, feel free to reopen  if needed
